node/modules/storageminer.go

```diff
@@ -596,15 +575,32 @@ func BasicDealFilter(user dtypes.StorageDealFilter) func(onlineOk dtypes.Conside
 	}
 }
 
+func DagStoreWrapper(
+	ds dtypes.MetadataDS,
+	r repo.LockedRepo,
+	pieceStore dtypes.ProviderPieceStore,
+	rpn retrievalmarket.RetrievalProviderNode,
+) (mktdagstore.DagStoreWrapper, error) {
+	dagStoreDir := filepath.Join(r.Path(), "dagstore")
+	dagStoreDS := namespace.Wrap(ds, datastore.NewKey("/dagstore/provider"))
+	cfg := mktdagstore.MarketDAGStoreConfig{
+		TransientsDir: filepath.Join(dagStoreDir, "transients"),
+		IndexDir:      filepath.Join(dagStoreDir, "index"),
+		Datastore:     dagStoreDS,
+	}
+	mountApi := mktdagstore.NewLotusMountAPI(pieceStore, rpn)
+	return mktdagstore.NewDagStoreWrapper(cfg, mountApi)
+}
+
```

Comment:
- I think Lotus should be owning and instantiating the DAG store via dependency injection. 
    - Having markets own this important part of the system makes it escape dependency injection, and it also makes it way harder for other parts of the system (e.g. CLI, JSON-RPC) to interact with the DAG store.
    - The DAG store should be a top-level Lotus component on the markets node.
- We can get rid of the `MarketDAGStoreConfig` struct, and just use `dagstore.Config`.
- The DAG store transients and index dir should be provided by Lotus configuration, falling back to defaults if unset. It's very likely that users will want to set these paths manually.
- We need to create a Lotus config section for DAG store parameters (including throttling levels!) 
- I _think_ that the Lotus mount should live in lotus because it's Lotus-specific (miner APIs are private and not standardised AFAIK), so this component is not reusable outside of Lotus.

Reply:
@dirkmc is working on PRs for these.

Reply:
PR to move lotus-specific code to lotus:
- [x] go-fil-markets: https://github.com/filecoin-project/go-fil-markets/pull/578
- [ ] lotus

Reply:
@raulk please take a look at the above two PRs

I haven't made many changes to the code that is copied over the lotus. In particular I've left `MarketDAGStoreConfig` in place. I'm not sure if it makes sense to replace it with `dagstore.Config` because if we use the intermediate struct then we can hide the details of the DAG store from the rest of the codebase. eg the registry and failure channel are created inside `NewDagStoreWrapper`.

Let's continue the discussion in the PR at https://github.com/filecoin-project/lotus/pull/6793